shuup/front/forms/product_list_modifiers.py

```diff
@@ -182,3 +186,75 @@ def get_filters(self, request, data):
         categories = data.get("categories")
         if categories:
             return Q(shop_products__categories__in=list(categories))
+
+
+class ProductPriceFilter(SimpleProductListModifier):
+    is_active_key = "filter_products_by_price"
+    is_active_label = _("Filter products by price")
+    ordering_key = "filter_products_by_price_ordering"
+    ordering_label = _("Ordering for filter by price")
+    range_min_key = "filter_products_by_price_range_min"
+    range_max_key = "filter_products_by_price_range_max"
+    range_size_key = "filter_products_by_price_range_size"
+
+    def get_fields(self, request, category=None):
+        if not category:
+            return
+
+        configuration = get_configuration(request.shop, category)
+
+        min_price = configuration.get(self.range_min_key)
+        max_price = configuration.get(self.range_max_key)
+        range_size = configuration.get(self.range_size_key)
+        if not (min_price and max_price and range_size):
+            return
+
+        choices = [(None, "-------")] + get_price_ranges(
+            request.shop, min_price, max_price, range_size)
+        return [
+            ("price_range", forms.ChoiceField(required=False, choices=choices, label=_("Price"))),
+        ]
+
+    def filter_products(self, request, products, data):
+        selected_range = data.get("price_range")
+        if not selected_range:
+            return products
+
+        min_price, max_price = selected_range.split("-", 1)
+        min_price_value = decimal.Decimal(min_price or 0)
+        max_price_value = decimal.Decimal(max_price or 0)
+        filtered_products = []
+        for product in products:
+            price_value = product.get_price(request).amount.value
+            if price_value >= min_price_value and (max_price == "" or price_value < max_price_value):
+                filtered_products.append(product)
+        return filtered_products
+
+    def get_admin_fields(self):
+        default_fields = super(ProductPriceFilter, self).get_admin_fields()
+        return default_fields + [
+            (self.range_min_key, forms.IntegerField(label=_("Price range minimum"), required=False)),
+            (self.range_max_key, forms.IntegerField(label=_("Price range maximum"), required=False)),
+            (self.range_size_key, forms.IntegerField(label=_("Price range size"), required=False)),
+        ]
+
+
+def get_price_ranges(shop, min_price, max_price, range_step):
+    if range_step == 0:
```

Comment:
This if `range_step` is `< 0` this breaks, unless you demand positive integers in form.